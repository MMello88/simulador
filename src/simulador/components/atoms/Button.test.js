import React from 'react';
import { render, fireEvent } from 'test-utils';

import Button, { ButtonColors, ButtonsVariants } from 'simulador/components/atoms/Button';


test('renders a text', () => {
  const { getByText } = render(
      <Button>Click here</Button>
  );

  expect(getByText('Click here')).toBeInTheDocument();
});

test('trigger event on click', () => {
  const handleClick = jest.fn();

  const { getByRole } = render(<Button onClick={handleClick} />);

  fireEvent.click(getByRole('button'));

  expect(handleClick).toBeCalled();
});


test.each(Object.values(ButtonColors).map(item => [item]))('render with collor %s', (color) => {
  const {asFragment} = render(<Button color={color}>Click here</Button>);
  
  expect(asFragment()).toMatchSnapshot();
});

test.each(Object.values(ButtonsVariants).map(item => [item]))('render with collor %s', (variant) => {
  const {asFragment} = render(<Button variant={variant}>Click here</Button>);
  
  expect(asFragment()).toMatchSnapshot();
});

test('renders with color primary and variant outlined', () => {
  const {asFragment} = render(<Button variant="outlined">Click here</Button>);
  
  expect(asFragment()).toMatchSnapshot();
})

test('renders with color primary and variant link', () => {
  const {asFragment} = render(<Button variant="link">Click here</Button>);
  
  expect(asFragment()).toMatchSnapshot();
})
