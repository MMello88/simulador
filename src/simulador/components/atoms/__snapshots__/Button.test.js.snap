// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with collor default 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  border: 2px solid;
  color: #212121;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

<button
    class="c0"
    color="default"
    type="button"
  >
    Click here
  </button>
</DocumentFragment>
`;

exports[`render with collor default 2`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  border: 2px solid;
  color: #212121;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

<button
    class="c0"
    color="default"
    type="button"
  >
    Click here
  </button>
</DocumentFragment>
`;

exports[`render with collor link 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  border: 2px solid;
  color: #212121;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #757575;
  padding-left: 0;
  padding-right: 0;
}

.c1:hover:enabled {
  background-color: transparent;
  border-color: transparent;
}

<button
    class="c0 c1"
    color="default"
    type="button"
  >
    Click here
  </button>
</DocumentFragment>
`;

exports[`render with collor outlined 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  border: 2px solid;
  color: #212121;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c1 {
  background-color: transparent;
  color: #212121;
}

.c1:hover:enabled {
  background-color: transparent;
}

<button
    class="c0 c1"
    color="default"
    type="button"
  >
    Click here
  </button>
</DocumentFragment>
`;

exports[`renders with color primary and variant link 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  border: 2px solid;
  color: #212121;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #757575;
  padding-left: 0;
  padding-right: 0;
}

.c1:hover:enabled {
  background-color: transparent;
  border-color: transparent;
}

<button
    class="c0 c1"
    color="default"
    type="button"
  >
    Click here
  </button>
</DocumentFragment>
`;

exports[`renders with color primary and variant outlined 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  border: 2px solid;
  color: #212121;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c1 {
  background-color: transparent;
  color: #212121;
}

.c1:hover:enabled {
  background-color: transparent;
}

<button
    class="c0 c1"
    color="default"
    type="button"
  >
    Click here
  </button>
</DocumentFragment>
`;
